src/h3lib/lib/iterators.c

```diff
@@ -0,0 +1,271 @@
+|   6 |          97 | 0 | 0 | 0 | 0 | 0 | 2 | ... |
+                                       \
+                                        first nonzero digit
+
+Iteration is normal (same as a hexagon) if we're not on the first
+nonzero digit.
+
+iterStepChild ->
+                            parent res      child res
+                           /               /
+| res | base cell # | 1 | 2 | 3 | 4 | 5 | 6 | ... |
+|-----|-------------|---|---|---|---|---|---|-----|
+|   6 |          97 | 0 | 0 | 0 | 0 | 0 | 3 | ... |
+                                       \
+                                        first nonzero digit
+
+...
+
+iterStepChild ->
+                            parent res      child res
+                           /               /
+| res | base cell # | 1 | 2 | 3 | 4 | 5 | 6 | ... |
+|-----|-------------|---|---|---|---|---|---|-----|
+|   6 |          97 | 0 | 0 | 0 | 0 | 0 | 6 | ... |
+                                       \
+                                        first nonzero digit
+
+iterStepChild ->
+
+We skip the `1` when we hit the next first nonzero digit.
+
+                            parent res      child res
+                           /               /
+| res | base cell # | 1 | 2 | 3 | 4 | 5 | 6 | ... |
+|-----|-------------|---|---|---|---|---|---|-----|
+|   6 |          97 | 0 | 0 | 0 | 0 | 2 | 0 | ... |
+                                   \
+                                    first nonzero digit
+
+iterStepChild ->
+
+We won't skip the `1` in a previous digit when it
+is no longer the first nonzero digit.
+
+                            parent res      child res
+                           /               /
+| res | base cell # | 1 | 2 | 3 | 4 | 5 | 6 | ... |
+|-----|-------------|---|---|---|---|---|---|-----|
+|   6 |          97 | 0 | 0 | 0 | 0 | 2 | 1 | ... |
+                                   \
+                                    first nonzero digit
+ */
+
+/*
+Initialize a Iter_Child struct representing the sequence giving
+the children of cell `h` at resolution `childRes`.
+
+At any point in the iteration, starting once
+the struct is initialized, Iter_Child.h gives the current child.
+
+Also, Iter_Child.h == H3_NULL when all the children have been iterated
+through, or if the input to `iterInitParent` was invalid.
+ */
+Iter_Child iterInitParent(H3Index h, int childRes) {
+    Iter_Child it;
+
+    it.parentRes = H3_GET_RESOLUTION(h);
+
+    if (childRes < it.parentRes || childRes > MAX_H3_RES || h == H3_NULL) {
+        return _null_iter();
+    }
+
+    it.h = _zero_index_digits(h, it.parentRes + 1, childRes);
+    H3_SET_RESOLUTION(it.h, childRes);
+
+    if (H3_EXPORT(h3IsPentagon)(it.h))
+        // The first nonzero digit skips `1` for pentagons.
+        // The "fnz" moves to the left as we count up from the child resolution
+        // to the parent resolution.
+        it.fnz = childRes;
+    else
+        // if not a pentagon, we can ignore "first nonzero digit" logic
+        it.fnz = -1;
```

Comment:
Please include curly braces here, especially since the interior block spans more than one line (with the comments)

Reply:
Hmm. Weird. I usually wouldn't write it that way. Anyway, fixed!